driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/pageobjects/internal/action/GuiElementCheckFieldAction.java

```diff
@@ -70,13 +73,34 @@ private void pCheckField(UiElementBase guiElement, Check check) {
                 }
             }
         }
-
         int prevTimeout = -1;
-
         if (useTimeout >= 0) {
             prevTimeout = overrides.setTimeout(useTimeout);
         }
 
+        /**
+         * It does not make sense to set both flags optional and collect!
+         *
+         * Because an optional assertion is always collected but without PageFactoryException
+         * the 'optional' is prioritised over 'collected'.
+         * In that case 'collected' will be ignored.
+         */
+
+        // Handling collected assertions
+        Assertion previousAssertionImpl = null;
+        if (check.collect() && !check.optional()) {
```

Comment:
Could have been simpified as
```java
if (check.optional()) {
  [...]
}
else if(check.collect()) {
  [...]
}
```
Maybe, if we'd like to ~punish~ mildly admonish senseless behaviour rewrite to
```java
if (check.collect() && check.optional()) {
  // Runtime.getRuntime().exec("format c:")
  [mild admonishment]
}
else if (check.collect()) {
  [...]
}
else if (check.optional()) {
  [...]
}
```